lib/media/streaming_engine.js

```diff
@@ -2000,6 +2045,35 @@ shaka.media.StreamingEngine = class {
    * @suppress {strictMissingProperties}
    */
   async fetch_(mediaState, reference, streamDataCallback) {
+    let op = null;
+    if (mediaState.segmentPrefetch) {
+      op = mediaState.segmentPrefetch.getPrefetchedSegment(reference);
+    }
+    if (!op) {
+      op = this.dispatchFetch_(
+          reference, mediaState.stream, streamDataCallback,
+      );
+    }
+
+    mediaState.operation = op;
+    const response = await op.promise;
+    mediaState.operation = null;
+    return response.data;
+  }
+
+  /**
+   * Fetches the given segment.
+   *
+   * @param {!shaka.extern.Stream} stream
+   * @param {(!shaka.media.InitSegmentReference|!shaka.media.SegmentReference)}
+   *   reference
+   * @param {?function(BufferSource):!Promise=} streamDataCallback
+   *
+   * @return {!shaka.net.NetworkingEngine.PendingRequest}
+   * @private
+   * @suppress {strictMissingProperties}
```

Comment:
Why?

Reply:
It was originally copied from fetch_() as this method was extracted from that.
I just tried remove it but got following err. Added instanceof check and working now.


```
[JSC_STRICT_INEXISTENT_UNION_PROPERTY] Property startTime not defined on all member types of (shaka.media.InitSegmentReference|shaka.media.SegmentReference)
  2091|           duration: reference.endTime - reference.startTime,
                                                          ^^^^^^^^^
```